### PR TITLE
Denmark extensions: Updates CPR number generator with checksum

### PR DESCRIPTION
**Pull request by MunroRaymaker**
Some older systems still use CPR number checksum validation. This updates the danish CPR number extension, to optionally create CPR numbers that can be validated against checksums.
Unit tests are also provided for the extension updates.

### CODE REVIEW

File: Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs [hunk 1]
```diff
@@ -8,19 +11,169 @@ public static class ExtensionsForDenmark
...
+               hasValidChecksum = GenerateChecksum(birthDate, p.Gender, individualNumber, out checksum);
+            } while (!hasValidChecksum);
+         }
+         else
+         {
+            checksum = string.Empty;
+            individualNumber = GenerateIndividualFourDigitNumber(r, p.Gender, p.DateOfBirth.Year);
+         }
+
+         var final = $"{birthDate}-{individualNumber}{checksum}";
```
Comment (NielsPilgaard) [inline Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs:67]: It would be nice to have a parameter to toggle this dash off.

**Review by NielsPilgaard — APPROVED** [review]
I find this change very helpful, and the logic looks good. I'm not a Bogus Contributor though, so I'm not sure my approval helps much 😅 

The current CPR implementation in Bogus doesn't generate valid CPR numbers, which is a bit counter-intuitive to me.

**Comment by MunroRaymaker** [comment]
Struggling to push a commit that resolve the conflicts. A lot has happened in master since I created this pullrequest.
I getting this error when I try a git push:

> JRN@DK-PC5CG3070Q8R MINGW64 /c/src/github/Bogus_branch_with_changes_to_danish_cpr/Source (dk-extensions-cpr-number)
> $ git push
> fatal: The current branch dk-extensions-cpr-number has no upstream branch.
> To push the current branch and set the remote as upstream, use
> 
>     git push --set-upstream origin dk-extensions-cpr-number
> 
> To have this happen automatically for branches without a tracking
> upstream, see 'push.autoSetupRemote' in 'git help config'.
> 

Don't understand why I need to create an upstream branch. Thought that was already there.

**Comment by NielsPilgaard** [comment]
It does look like the upstream branch is present on your repo. Perhaps it would be easiest to start over on a fresh branch?

**Comment by bchavez** [comment]
Merge conflict should be resolved;
The crux of the problem is a conversion to file-scoped namespaces;
If the build passes; we will merge. Thanks again for the PR.

**Comment by bchavez** [comment]
@NielsPilgaard @MunroRaymaker if you could have one last look to ensure the rebase was successful that would be great.
Lots of thanks,
Brian

File: Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs [hunk 2]
```diff
@@ -8,18 +11,168 @@ public static class ExtensionsForDenmark
...
+         to = 9999;
+      }
+      else if (2037 <= year && year <= 2057)
+      {
+         from = 5000;
+         to = 9999;
+      }
+      else
+      {
+         throw new ArgumentOutOfRangeException(nameof(year), $"{nameof(year)} must be between 1854 and 2039.");
```
Comment (NielsPilgaard) [inline Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs:105]: The else if above suggest the range ends at 2057, which year is correct?

File: Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs [hunk 3]
```diff
@@ -8,18 +11,168 @@ public static class ExtensionsForDenmark
...
+         to = 999;
+      }
+      else if (2037 <= year && year <= 2057)
+      {
+         from = 500;
+         to = 999;
+      }
+      else
+      {
+         throw new ArgumentOutOfRangeException(nameof(year), $"{nameof(year)} must be between 1854 and 2039.");
```
Comment (NielsPilgaard) [inline Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs:145]: The else if above suggest the range ends at 2057, which year is correct?

**Review by NielsPilgaard — APPROVED** [review]
Looks good to me still, everything is intact after the merge. 
I spotted and commented on 2 minor issues, but they're not blocking in my opinion.

File: Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs (hunk 1 above)
Comment (bchavez) [reply Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs]: The `includeDash` parameter has been added in the latest commit

File: Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs (hunk 2 above)
Comment (bchavez) [reply Source/Bogus/Extensions/Denmark/ExtensionsForDenmark.cs]: This should be fixed in the latest commit. Please have a look.

**Review by NielsPilgaard — APPROVED** [review]
LGTM, thanks for adding the include dash parameter 👍

**Comment by MunroRaymaker** [comment]
Thanks guys for help with the PR 🥇

**Pull request merged**